### PR TITLE
Added Motion-UI support to ResponsiveToggle #7940

### PR DESCRIPTION
**Pull request by DominikPieper**
Add support for Motion-UI to the ResponsiveToggle component, using data-animate=".." to configure the animations.

### CODE REVIEW

**Comment by zurbchris** [comment]
Awesome! We'll have to do some testing on this, but thanks for knocking this one out.

**Comment by rafibomb** [comment]
@nachtwandler We want to add this into the 6.2 RC - are you able to clear the merge conflicts so we can add it for testing?

File: js/foundation.responsiveToggle.js [hunk 1]
```diff
@@ -60,7 +77,7 @@ ResponsiveToggle.prototype._events = function() {
 
   $(window).on('changed.zf.mediaquery', this._update.bind(this));
 
-  this.$toggler.on('click.zf.responsiveToggle', this.toggleMenu.bind(this));
+  this.$toggler.on('click.zf.responsiveToggle', this.options.animate ? this.toggleMenuAnimated.bind(this) : this.toggleMenu.bind(this));
```
Comment (kball) [inline js/foundation.responsiveToggle.js]: I'd rather have this decision live within toggleMenu.  As it is, we end up with duplicated logic between toggleMenuAnimated and toggleMenu.  Lets instead have the inner part of toggleMenu check for this.options.animate and decide how to do the toggle.

File: js/foundation.responsiveToggle.js [hunk 2]
```diff
@@ -98,6 +115,30 @@ ResponsiveToggle.prototype.toggleMenu = function() {
...
+    }
+    else {
+      Foundation.Motion.animateOut(this.$targetMenu, this.animationOut);
+    }
+
+    /**
+     * Fires when the element attached to the tab bar toggles.
+     * @event ResponsiveToggle#toggled
+     */
+    this.$element.trigger('toggled.zf.responsiveToggle');
```
Comment (kball) [inline js/foundation.responsiveToggle.js]: Does it make sense to wait to trigger this until the animation is complete?

**Comment by nachtwandler** [comment]
@rafibomb Nice to hear. Because of my projects I think the first time I can take a deeper look is at the weekend. Maybe after the full rewrite to ES6 it's easier and faster to re implement it from zero rather than only clear the merge conflicts. Also I can take @kball's notes into account. ( @kball thanks for the notes)

**Comment by kball** [comment]
@nachtwandler do you want to take another look at this now that the ES6 rewrite is in?  Thanks

**Comment by nachtwandler** [comment]
@kball Hi, yes i will take a look as soon as possible. Sorry for the long time but cause of my current projects I had not the time so far.

**Comment by nachtwandler** [comment]
@kball Hi, this should work now. Sorry for the strange commits, I struggled a bit with the git branch functionality.

**Comment by kball** [comment]
This looks good to me.  However it is currently submitted against the develop branch, which is currently only accepting bugfixes.  @nachtwandler can you resubmit it against v6.3?

**Comment by kball** [comment]
Closing this one in favor of the one that got merged into 6.3

**Comment by pine3ree** [comment]
@nachtwandler 
Hello, I am trying - unsuccesfully -  to use this pr to have a simple slide down effect. btw sliding to/from left/right works. To achieve that i had to disable the responsiveToggle + motion-ui and use plain jquery slideDow/Up methods. Any hint?
kind regards

**Pull request closed without merging**